proto/messages.proto

```diff
@@ -14,9 +14,13 @@ message GenericMessage {
     ClientAction clientAction = 9;
     Calling calling = 10;
     Asset asset = 11;
-    MsgDeleted deleted = 12;
+    MessageHide hidden = 12;
     Location location = 13;
+    MessageDelete deleted = 14;
   }
+
+  optional string version_id = 15;
+  optional int32 version_number = 16;
```

Comment:
It might make more sense to extract this in a separate message to ensure that both fields are always set if the message has a version